src/UI/Art/SpriteArtist/SpriteMapView.java

```diff
@@ -192,18 +192,31 @@ public void animate( GameEventQueue newEvents )
    */
   private void loadNextEventAnimation()
   {
+    // Here are the fog-drawing rules. If there are:
+    //   zero humans - spectating - draw everything the current player sees.
+    //   one human - player vs ai - draw everything the human player could see.
+    //   2+ humans - player vs player - draw what the current player sees, IFF the player is human.
+    GameMap gameMap = myGame.activeCO.myView; // Start by assuming zero humans.
+    int numHumans = countHumanPlayers(myGame);
+    if( 1 == numHumans )
+    {
+      // Since there is only one human, always use the human's vision to determine what is drawn.
+      gameMap = getHumanPlayerMap(myGame);
+    }
+    // Don't let humans peek in on what the AI is doing.
+    boolean drawFogEverywhere = myGame.isFogEnabled() && (numHumans > 1) && (myGame.activeCO.isAI());
+
     // Keep pulling events off the queue until we get one we can draw.
     while( null == currentAnimation && !eventsToAnimate.isEmpty() )
     {
       GameEvent event = eventsToAnimate.peek();
-      if( null != event )
+      currentAnimation = event.getEventAnimation( this );
+      boolean isEventHidden = !(null == event.getStartPoint()) && gameMap.isLocationFogged(event.getStartPoint()) && gameMap.isLocationFogged(event.getEndPoint());
```

Comment:
This would crash if CO abilities had animations.
Perhaps null checking should be done in the XYCoord-based getters?

Reply:
Would it? `getStartPoint()` would return null, and when it compares positively with null here, it would be inverted to false, which means the whole chain of 'and's cannot succeed, and the subsequent calls to `getStartPoint()` would not trigger; `isEventHidden` would just be set to false. Or did I mess up the logic somhow?

Reply:
You're right. I'm wrong.
I guess whenever we actually have animations for CO abilities, we're gonna have locations for them to happen on as well... but I feel like maybe powers should always be animated? I dunno, probably kick that thought down the road.

Reply:
I agree that Commander Abilities should always be animated. The current behavior is to perform the animation if the event is not tied to a location.
That being said, a Commander Ability may trigger other events (unit spawns, etc) that _will_ have associated locations - so I think we have full flexibility to make things work however we want.
The "Commander Ability Activated!" splash or whatever should at least be shown.